contrib/src/main/scala/akka/stream/contrib/Valve.scala

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib
+
+import akka.NotUsed
+import akka.stream.stage.{GraphStageLogic, GraphStageWithMaterializedValue, InHandler, OutHandler}
+import akka.stream._
+import akka.stream.contrib.SwitchMode.{Close, Open}
+
+import scala.concurrent.{Future, Promise}
+
+/**
+  * Pause/ Resume a Flow
+  */
+trait ValveSwitch {
+  /**
+    * Change the state of the valve
+    *
+    * @param mode expected mode to switch on
+    * @return A future that completes with true if the mode did change and false if it already was in the requested mode
+    */
+  def flip(mode: SwitchMode): Future[Boolean]
+}
+
+
+object Valve {
+  /**
+    * Factory for [[Valve]] instances.
+    *
+    * @tparam A input and output type
+    * @return [[Valve]] Opened instance
+    *
+    */
+  def apply[A](): Valve[A] = new Valve[A](SwitchMode.Open)
+
+  /**
+    * Factory for [[Valve]] instances.
+    *
+    * @tparam A input and output type
+    * @return [[Valve]] instance
+    *
+    */
+  def apply[A](mode: SwitchMode): Valve[A] = new Valve[A](mode)
+
+  /**
+    * Java API: Factory for [[Valve]] instances.
+    *
+    * @tparam A input and output type
+    * @return [[Valve]] Opened instance
+    */
+  def create[A](): Valve[A] = new Valve[A](SwitchMode.Open)
+
+  /**
+    * Java API: Factory for [[Valve]] instances.
+    *
+    * @tparam A input and output type
+    * @return [[Valve]] instance
+    */
+  def create[A](mode: SwitchMode): Valve[A] = new Valve[A](mode)
```

Comment:
? I dont get your comment it is for the ***Java API***. as suggested @johanandren  i provided a factory to instanciate the Valve.